src/Uno.UI/UI/Xaml/Window.cs

```diff
@@ -200,16 +216,16 @@ public void Activate()
 			// for compatibility with WinUI we set the first activated
 			// as Current #8341
 			_current ??= this;
-
-			InternalActivate();
-
-			OnActivated(CoreWindowActivationState.CodeActivated);
+			_wasEverActivated = true;
 
 			// Initialize visibility on first activation.
 			Visible = true;
+			OnActivated(CoreWindowActivationState.CodeActivated);
```

Comment:
This feels the wrong way around - I would expect the platform activation (ie ActivatePartial) to happen first, before the public Activated event is raised (which is inside the OnActivated method)